include/crocoddyl/core/control-base.hpp

```diff
@@ -0,0 +1,187 @@
+///////////////////////////////////////////////////////////////////////////////
+// BSD 3-Clause License
+//
+// Copyright (C) 2021, LAAS-CNRS, University of Edinburgh, INRIA, University of Trento
+// Copyright note valid unless otherwise stated in individual files.
+// All rights reserved.
+///////////////////////////////////////////////////////////////////////////////
+
+#ifndef CROCODDYL_CORE_CONTROL_BASE_HPP_
+#define CROCODDYL_CORE_CONTROL_BASE_HPP_
+
+#include <vector>
+#include <string>
+#include <stdexcept>
+
+#include "crocoddyl/core/fwd.hpp"
+#include "crocoddyl/core/mathbase.hpp"
+#include "crocoddyl/core/utils/exception.hpp"
+
+namespace crocoddyl {
+
+/**
+ * @brief Abstract class for the control trajectory parametrization
+ *
+ * The control trajectory is a function of the (normalized) time.
+ * Normalized time is between 0 and 1, where 0 represents the beginning of the time step,
+ * and 1 represents its end.
+ * The trajectory depends on the control parameters p, whose size may be larger than the
+ * size of the control inputs u.
+ *
+ */
+template <typename _Scalar>
+class ControlParametrizationModelAbstractTpl {
+ public:
+  EIGEN_MAKE_ALIGNED_OPERATOR_NEW
+
+  typedef _Scalar Scalar;
+  typedef MathBaseTpl<Scalar> MathBase;
+  typedef ControlParametrizationDataAbstractTpl<Scalar> ControlParametrizationDataAbstract;
+  typedef typename MathBase::VectorXs VectorXs;
+  typedef typename MathBase::MatrixXs MatrixXs;
+
+  /**
+   * @brief Initialize the control dimensions
+   *
+   * @param[in] nx   Dimension of control
+   * @param[in] np   Dimension of control parameters
+   */
+  ControlParametrizationModelAbstractTpl(const std::size_t nu, const std::size_t np);
+  virtual ~ControlParametrizationModelAbstractTpl();
+
```

Comment:
Cosmetic change that I will handle it (don't want to bother you @andreadelprete).

Let's put an order (an assert it in each respectively case) in the declaration. The order should be `calc`, `calcDiff`, `createData`, `checkData`, `convertBounds`, `multiplyByJacobian`, and `multiplyJacobianTransposeBy`.